graph-proxy/src/graphql/workflows.rs

```diff
@@ -445,3 +445,419 @@ impl WorkflowsQuery {
         Ok(connection)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use async_graphql::{Name, Value};
+    use indexmap::IndexMap;
+    use url::Url;
+
+    use super::*;
+    use crate::graphql::root_schema_builder;
+
+    #[tokio::test]
+    async fn single_workflow_query() {
+        let workflow_name = "numpy-benchmark-wdkwj";
+        let proposal_code = "mg";
+        let proposal_number = 36964;
+        let number = 1;
+        let namespace = format!("{}{}-{}", proposal_code, proposal_number, number);
+
+        let server = httpmock::MockServer::start();
+        let mut response_file_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        response_file_path.push("test-resources");
+        response_file_path.push("get-workflow-wdkwj.json");
+        let workflow_endpoint = server.mock(|when, then| {
+            when.method(httpmock::Method::GET)
+                .path(format!("/api/v1/workflows/{}/{}", namespace, workflow_name));
+            then.status(200)
+                .header("content-type", "application/json")
+                .body(std::fs::read_to_string(response_file_path).unwrap());
+        });
+
+        let token = None::<Authorization<Bearer>>;
+        let argo_server_url = Url::parse(&server.base_url()).unwrap();
+        let schema = root_schema_builder()
+            .data(ArgoServerUrl(argo_server_url))
+            .data(token)
+            .finish();
```

Comment:
Can inline the `token` here as it's not used anywhere else

```suggestion
        let argo_server_url = Url::parse(&server.base_url()).unwrap();
        let schema = root_schema_builder()
            .data(ArgoServerUrl(argo_server_url))
            .data(None::<Authorization<Bearer>>)
            .finish();
```

Reply:
Ah good point, will make that change in all the tests